src/TwigComponent/src/HasAttributesTrait.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\UX\TwigComponent;
+
+use Symfony\UX\LiveComponent\Attribute\LiveProp;
+use Symfony\UX\TwigComponent\Attribute\PostMount;
+use Symfony\UX\TwigComponent\Attribute\PreMount;
+
+/**
+ * @author Kevin Bond <kevinbond@gmail.com>
+ *
+ * @experimental
+ */
+trait HasAttributesTrait
+{
+    #[LiveProp(hydrateWith: 'hydrateAttributes', dehydrateWith: 'dehydrateAttributes')]
+    public ComponentAttributes $attributes;
+
+    #[PreMount]
+    public function preMountAttributes(array $data): array
+    {
+        $data['_attributes'] = $data['attributes'] ?? [];
+
+        unset($data['attributes']);
+
+        return $data;
+    }
+
+    #[PostMount(priority: -1000)]
+    public function mountAttributes(array $data): array
+    {
+        $attributes = array_merge($data['_attributes'], $data);
+        unset($attributes['_attributes']);
+
+        $this->attributes = new ComponentAttributes($attributes);
```

Comment:
Should we do some error checking here to ensure these values are all scalar? If they are not scalar, should I throw an exception here or return the array including this value and let the "Unable to write prop to component..." exception be thrown?

Reply:
My gut says to start MORE strict. It feels weird to add the check here, but I think we should: loop over them, and check for scalar values. If not scalar, throw an exception.

But the error message is key, because there are two possibilities:

A) I am trying to set an attribute, but I accidentally passed something weird
B) I am trying to pass a real piece of data, but I typo'ed the data or the property name, so there's a mismatch.

Probably (B) is more common. But if we throw the exception here, we can address both in the same message.

Reply:
Added in 11b6989802cfa65f0f1a311ece1b94c9632cb28e.